test/org/centos/contra/Infra/Utils.groovy

```diff
@@ -0,0 +1,347 @@
+import spock.lang.*
+import com.homeaway.devtools.jenkins.testing.JenkinsPipelineSpecification
+
+import org.yaml.snakeyaml.Yaml
+
+public class UtilsSpec extends JenkinsPipelineSpecification {
+    @Shared
+    def config = new HashMap<String, ?>()
+
+    @Shared
+    def infraUtils
+
+    def setup() {
+        infraUtils = loadPipelineScriptForTest("../../src/org/centos/contra/Infra/Utils.groovy")
+        explicitlyMockPipelineStep('dir')
```

Comment:
These steps should be mocked by Jenkins-Spock. By adding https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/ and  https://jenkins.io/doc/pipeline/steps/kubernetes/ to your pom.xml, these will be mocked for free. Then you can do ` 1 * getPipelineMock("writeFile")..`